pilot/pkg/proxy/envoy/v2/ads.go

```diff
@@ -429,6 +436,9 @@ func (s *DiscoveryServer) StreamAggregatedResources(stream ads.AggregatedDiscove
 				return err
 			}
 			nt.Metadata = model.ParseMetadata(discReq.Node.Metadata)
+			if nt.Metadata["IPAddresses"] != "" {
```

Comment:
look at how some other ISTIO_META_ stuff is done..
https://github.com/istio/istio/blob/master/install/kubernetes/helm/istio/templates/sidecar-injector-configmap.yaml#L157

this needs to be ISTIO_META_INSTANCE_IPS. 
So you have something like
model.ParseServiceNodeWithMetadata(discReq.Node.Id, model.ParseMetadata(discReq.Node.Metadata))

Also, move the metadata thing to model package please.. model/context.go, next to the node stuff.